server/src/main/java/org/apache/cassandra/sidecar/routes/cdc/StreamCDCSegment.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.cassandra.sidecar.common.utils.HttpRange;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.models.HttpResponse;
+import org.apache.cassandra.sidecar.routes.AbstractHandler;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.FileStreamer;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.getIdxFileName;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.isLogFile;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.isValid;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.parseIndexFile;
+
+/**
+ * Provides REST endpoint for streaming cdc commit logs.
+ */
+@Singleton
+public class StreamCDCSegment extends AbstractHandler<String>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(StreamCDCSegment.class);
+
+    private final FileStreamer fileStreamer;
+    private final CDCLogCache cdcLogCache;
+
+    @Inject
+    public StreamCDCSegment(final InstanceMetadataFetcher metadataFetcher,
+                            final FileStreamer fileStreamer,
+                            final CDCLogCache cdcLogCache,
+                            final ExecutorPools executorPools,
+                            final CassandraInputValidator validator)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.fileStreamer = fileStreamer;
+        this.cdcLogCache = cdcLogCache;
+    }
+
+    @Override
+    protected void handleInternal(RoutingContext context,
+                                  HttpServerRequest httpRequest,
+                                  String host,
+                                  SocketAddress remoteAddress,
+                                  String segment)
+    {
+        final InstanceMetadata instance = metadataFetcher.instance(host);
+        final String cdcDir = instance.cdcDir();
+        final String idxFileName = getIdxFileName(segment);
+        final File segmentFile = new File(cdcDir, segment);
+        final File idxFile = new File(cdcDir, idxFileName);
+        Promise<Void> promise = Promise.promise();
+        streamSegment(context, segmentFile, idxFile, promise);
+        // Touch the files at the end of the request
+        // If the file exists in cache, its expiry is extended; otherwise, the cache is not changed.
+        promise.future().onSuccess(res -> cdcLogCache.touch(segmentFile, idxFile));
+    }
+
+    /**
+     * Validate file existence and hardlink the cdc commit log file and the index file.
+     */
+    private void streamSegment(RoutingContext ctx,
+                               File segmentFile,
+                               File indexFile,
+                               Promise<Void> promise)
+    {
+        final HttpResponse response = new HttpResponse(ctx.request(), ctx.response());
```

Comment:
Remove the unnecessary `final` modifiers. Generally only declare `final` for instance variables. Please double check the usage in the other files too.